invenio/modules/oauthclient/contrib/cern.py

```diff
@@ -0,0 +1,239 @@
+    signup_handler=dict(
+        info="invenio.modules.oauthclient.contrib.cern:account_info",
+        setup="invenio.modules.oauthclient.contrib.cern:account_setup",
+        view="invenio.modules.oauthclient.handlers:signup_handler",
+    ),
+    params=dict(
+        request_token_params={
+                "resource": "datadev00.cern.ch",
+                "scope": "Name Email Bio Groups",
+        },
+        base_url="https://oauth.web.cern.ch/",
+        request_token_url=None,
+        access_token_url="https://oauth.web.cern.ch/OAuth/Token",
+        access_token_method="POST",
+        authorize_url="https://oauth.web.cern.ch/OAuth/Authorize",
+        app_key="CERN_APP_CREDENTIALS",
+        content_type="application/json",
+    )
+)
+"""CERN Remote Application."""
+
+REMOTE_SANDBOX_APP = copy.deepcopy(REMOTE_APP)
+"""CERN Sandbox Remote Application."""
+
+REMOTE_SANDBOX_APP["params"].update(dict(
+    base_url="https://oauth.web.cern.ch/",
+    access_token_url="https://oauth.web.cern.ch/OAuth/Token",
+    authorize_url="https://oauth.web.cern.ch/OAuth/Authorize",
+))
+
+def fetch_groups(groups):
+    groups = [group for group in groups if group not in
+              CFG_EXTERNAL_AUTH_HIDDEN_GROUPS]
+    for regexp in CFG_EXTERNAL_AUTH_HIDDEN_GROUPS_RE:
+        for group in groups:
+            if regexp.match(group):
+               groups.remove(group)
+
+    return dict(map(lambda x: (x, '@' in x and x + ' (Mailing list)'
+                                   or x + ' (Group)'), groups))
+
+def get_dict_from_response(response):
+    result = {}
+    for i in response.json():
+        k = str(i['Type'])
+        if not result.get(k, None):
+            result[k] = []
+        result[k].append(str(i['Value']))
+
+    return result
+
+def account_info(remote, resp):
+    """Retrieve remote account information used to find local user."""
+    from flask import current_app
+    current_app.logger.info(remote)
+    current_app.logger.info(resp)
+
+    # Query CERN Resources to get user info and groups
+    url = 'https://oauthresource.web.cern.ch/api/Me'
+    headers =  {'Authorization': 'Bearer '+resp['access_token']}
+    response = requests.get(url, headers=headers)
+    res = {}
+    res = get_dict_from_response(response)
+    email = res['http://schemas.xmlsoap.org/claims/EmailAddress']
+    nickname = res['http://schemas.xmlsoap.org/claims/Firstname']
+    return dict(email=email, nickname=nickname, resp = res)
+
+def account_setup(remote, token):
+    """Perform additional setup after user have been logged in."""
+    from invenio.modules.accounts.models import User, UserEXT, Usergroup
+    from invenio.ext.sqlalchemy import db
+    from ..handlers import token_session_key
+
+    from sqlalchemy.orm.exc import MultipleResultsFound, NoResultFound
+
+    url = 'https://oauthresource.web.cern.ch/api/Me'
+    headers =  {'Authorization': 'Bearer '+token.token()[0]}
+    response = requests.get(url, headers=headers)
```

Comment:
I think you can actually use the flask-oauthlib remote app to make the requests with. See https://flask-oauthlib.readthedocs.org/en/latest/client.html#invoking-remote-methods